packages/web/src/services/FileDownloadServiceWeb.ts

```diff
@@ -1,23 +1,113 @@
+import axios from "axios";
+import StreamSaver from "streamsaver";
+
 import {
     FileDownloadService,
     DownloadResult,
     FileInfo,
     DownloadResolution,
-    HttpServiceBase,
+    FileStorageServiceBase,
 } from "../../../core/services";
 
-export default class FileDownloadServiceWeb extends HttpServiceBase implements FileDownloadService {
+export default class FileDownloadServiceWeb
+    extends FileStorageServiceBase
+    implements FileDownloadService {
     isFileSystemAccessible = false;
 
     public async download(fileInfo: FileInfo): Promise<DownloadResult> {
+        if (fileInfo.path.endsWith(".zarr")) {
+            const downloadResult = await this.handleZarrFile(fileInfo);
+            if (downloadResult) {
+                return downloadResult;
+            }
+        }
+
+        return this.downloadFile(fileInfo);
+    }
+
+    private async handleZarrFile(fileInfo: FileInfo): Promise<DownloadResult | null> {
+        if (this.isS3Url(fileInfo.path)) {
+            return await this.downloadS3Directory(fileInfo);
+        }
+
+        const localDownloadResult = this.isLocalPath(fileInfo.path, fileInfo);
+        return localDownloadResult; // This will be either a DownloadResult or null
+    }
+
+    private isLocalPath(filePath: string, fileInfo: FileInfo): DownloadResult | null {
+        const uriPattern = /^(https?|ftp):\/\/|^[a-zA-Z]:\\/;
+        const isLocal = filePath.startsWith("file://") || !uriPattern.test(filePath);
```

Comment:
I don't think the path necessarily needs to start with `file://` ex. `/allen/aics/a/path/to/a/file` is how most of our paths look; unless "file://" is automatically added somewhere else?

Reply:
I wonder if instead of checking if it is a local file here we could:

For web, throw an error / alert when the file isn't on S3 since we don't support any other locations with a message like "The .zarr file located at <blah> is not accessible to this application. BioFile Finder currently only supports S3 locations & they must be accessible to you the user from this computer."

For desktop, if S3 do the download per usual. If not, try to access it using `fs` and if it works then just do the download. If that also doesn't work, then throw an error saying it isn't accessible to this application and provide the error we get from `fs`.